frontend/src/store/app/reducers.ts

```diff
@@ -31,6 +34,10 @@ export function appReducer(state = initialState, action: AppAction): AppState {
       // TODO: Redux recommends **against** using new date inside reducers; instead
       // it should come from the code which creates the action to dispatch.
       return { ...state, backendHeartbeat: new Date() };
+    case BACKEND_CONNECTION_DOWN:
+      return { ...state, clock: new Date(), backendConnection: action.backendConnection };
```

Comment:
Could we just set `backendConnection: false` for `BACKEND_CONNECTION_DOWN` and `backendConnection: true` for BACKEND_CONNECTION_UP`?